scvi/module/_jaxvae.py

```diff
@@ -0,0 +1,144 @@
+from typing import NamedTuple, Optional
+
+import jax
+import jax.numpy as jnp
+import numpyro.distributions as dist
+from flax import linen as nn
+from flax.linen.initializers import variance_scaling
+
+from scvi import REGISTRY_KEYS
+
+
+class Dense(nn.Dense):
+    def __init__(self, *args, **kwargs):
+        # https://github.com/google/jax/blob/ab15db7d8658825afa9709df46f0dea76688d309/jax/_src/nn/initializers.py#L169
+        # sqrt 5 comes from PyTorch
+        kernel_init = variance_scaling(5.0, "fan_in", "uniform")
+        bias_init = variance_scaling(5.0, "fan_in", "uniform", in_axis=-1)
+        kwargs.update({"kernel_init": kernel_init, "bias_init": bias_init})
+        super().__init__(*args, **kwargs)
+
+
+class FlaxEncoder(nn.Module):
+    n_input: int
+    n_latent: int
+    n_hidden: int
+    dropout_rate: int
+    is_training: Optional[bool] = None
+
+    @nn.compact
+    def __call__(self, inputs, is_training):
+
+        is_training = nn.merge_param("is_training", self.is_training, is_training)
```

Comment:
from what i understand this is meant to avoid the dataclass `is_training` to not conflict w the arg in `__call__`. However, by making it an arg it seems like you want it to be always passed on `__call__`. Is there a point to having the dataclass attr?

Reply:
this is a good question -- let's address this in future PRs as I don't know!